src/lib/syncable_ledger/syncable_ledger.ml

```diff
@@ -49,7 +49,11 @@ module type S = sig
 
   val destroy : t -> unit
 
-  val new_goal : t -> root_hash -> unit
+  val new_goal : t -> root_hash -> [`Ignore | `Continue]
```

Comment:
can you instead return:

```ocaml
[`Repeat | `New]
```

or something like that. Ignore/Continue was a little confusing to me